tests/cleaning/quality_filter_test.py

```diff
@@ -0,0 +1,218 @@
+
+    @pytest.fixture(scope="class")
+    def all_texts(self, tweet_texts, long_text, bullets_text, ellipsis_text):
+        return tweet_texts + [long_text] + [bullets_text] + [ellipsis_text]
+
+    @pytest.fixture(scope="class")
+    def stop_words(self):
+        return set(
+            [
+                "er",
+                "jeg",
+                "det",
+                "du",
+                "ikke",
+                "at",
+                "en",
+                "og",
+                "har",
+                "vi",
+                "til",
+                "på",
+                "hvad",
+                "mig",
+                "med",
+                "de",
+                "for",
+                "den",
+                "så",
+                "der",
+                "dig",
+                "han",
+                "kan",
+                "af",
+            ]
+        )
+
+    @pytest.fixture(scope="class")
+    def quality_filter(self):
+        return QualityFilter()
+
+    def test_good_stop_words(self, quality_filter, stop_words):
+        assert (
+            quality_filter.stop_word(
+                quality_filter.nlp("jeg er glad"), n=2, stop_words=stop_words
+            )
+            is True
+        )
+
+    def test_bad_stop_words(self, quality_filter, stop_words):
+        assert (
+            quality_filter.stop_word(
+                quality_filter.nlp("56789okd23456789098765sds"),
+                n=1,
+                stop_words=stop_words,
+            )
+            is False
+        )
+
+    def test_good_bullets(self, quality_filter):
+        assert (
+            quality_filter.line_bullets_or_ellipsis(
+                quality_filter.nlp("56789okd23456789098765sds"),
+                max_p_bullets=0.5,
+                max_p_ellipsis=1,
+            )
+            is True
+        )
+
+    def test_bad_bullets_on_ellipsis(self, quality_filter, ellipsis_text):
+        assert (
+            quality_filter.line_bullets_or_ellipsis(
+                quality_filter.nlp(ellipsis_text), max_p_bullets=0.5, max_p_ellipsis=1
+            )
+            is False
+        )
+
+    def test_bad_bullets_on_bullets(self, quality_filter, bullets_text):
+        assert (
+            quality_filter.line_bullets_or_ellipsis(
+                quality_filter.nlp(bullets_text), max_p_bullets=0.5, max_p_ellipsis=1
+            )
+            is False
+        )
+
+    def test_good_ellipsis(self, quality_filter, ellipsis_text):
+        assert (
+            quality_filter.line_bullets_or_ellipsis(
+                quality_filter.nlp(ellipsis_text), max_p_bullets=1.0, max_p_ellipsis=0.5
+            )
+            is True
+        )
+
+    def test_bad_ellipsis_dots(self, quality_filter):
+        assert (
+            quality_filter.line_bullets_or_ellipsis(
+                quality_filter.nlp("jeg er glad..."),
+                max_p_bullets=1.0,
+                max_p_ellipsis=0.5,
+            )
+            is False
+        )
```

Comment:
This is basically a copy of the next test you can simplify it using `pytest.mark.parametrize `
```suggestion
    @pytest.mark.parametrize(
    "text,valid",
    [
        ("jeg er glad...", False),
        ("jeg er glad…", False),
    ],
    )
    def test_bad_ellipsis_dots(self, quality_filter, text, valid):
        assert (
            quality_filter.line_bullets_or_ellipsis(
                quality_filter.nlp("jeg er glad…"),
                max_p_bullets=1.0,
                max_p_ellipsis=0.5,
            )
            is valid
        )
```